### DISCUSSION TITLE
General Questions

### DISCUSSION THREAD

**Discussion by Sharpz7** [coder/coder#5685] [General]
Hi!

I am currently working through issues with setting coder up, and I have a few questions. I am running coder through the helm charts on k3s, and have successfully setup a workspace with the starter k8s template. (See https://github.com/coder/coder/issues/5683 for my main problem and the yaml of the helm chart manifest I am using)

I am hopefuly going to use a custom template based on: https://github.com/coder/coder/blob/bpmct/k8s-full/examples/templates/kubernetes-full/main.tf (i.e I want to have it run in kubernetes with access to docker through sysbox, and be able to run a custom image). I dont understand a few things about this template:

- ~I am very scared about volumes not being persistent. Is there anything else this template needs to ensure this will not happen? Are there any recommended methods for backing up with my setup?~ - I now know that as long as I don't delete the workspace the volume is persistent
- I also know there is a work around to support dev containers, how would I add that into this template not using gcp (Roughly doing this https://github.com/bpmct/coder-templates/blob/1287f9acac0635e8b725d0eadd59d3daabc600a3/gcp-devcontainer/main.tf#L110-L132)
- How do we access any development services from the browser? I couldn't get code-server to work (See https://files.mcaq.me/30578.jpg)

Update:
I have DinD working, by following the docs (Sysbox is not an option on k3s). I don't think volumes will work correctly though, as there is no volume to permanently store them, nor a method to access the host container for host volume mounts, If this is possible, could I be sent in the right direction to solve it?

**Comment by jsjoeio** [comment]
@bpmct is probably the best person to help here since he's the author of said custom template!

**Comment by matifali** [comment]
Hi, @Sharpz7 were you able to get it resolved?
